### PR TITLE
Refactor tabs theme 7.1.x

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #  

### Additional information (check all that apply):
 - [x] Bug fix

### Checklist:
 - [x] All relevant tags have been applied to this PR

### CODE REVIEW

File: projects/igniteui-angular/src/lib/core/styles/components/tabs/_tabs-theme.scss [hunk 1]
```diff
@@ -83,18 +89,66 @@
...
+    @if not($button-color) and $item-background {
+        $button-color: text-contrast($item-background);
+    }
+
     @if not($button-color) and $button-background {
         $button-color: text-contrast($button-background);
     }
 
+    @if not($button-color) and $item-background {
+        $button-background: rgba(text-contrast($item-background), 0);
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/tabs/_tabs-theme.scss]: Replace with transparent.

File: projects/igniteui-angular/src/lib/core/styles/components/tabs/_tabs-theme.scss [hunk 2]
```diff
@@ -83,18 +89,66 @@
...
         $button-hover-color: text-contrast($button-hover-background);
     }
 
+    @if not($button-hover-background) and $item-background {
+        $button-hover-color: text-contrast($item-background);
+    }
+
+    @if not($button-background) and $item-background {
+        @if (lightness($item-background) > 50) {
+            $button-background: rgba(0, 0, 0, 0);
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/tabs/_tabs-theme.scss]: Replace with transparent.

File: projects/igniteui-angular/src/lib/core/styles/components/tabs/_tabs-theme.scss [hunk 3]
```diff
@@ -139,16 +192,13 @@
 /// @requires --var
 @mixin igx-tabs($theme) {
     @include igx-root-css-vars($theme);
+    $item-min-width: 90px;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/tabs/_tabs-theme.scss]: Add max width to 360px. Wrap text if longer.

**Comment by simeonoff** [comment]
@sstoyanovIG can you help Marin resolve the failing tests?

**Comment by rkaraivanov** [comment]
@sstoyanovIG is this PR verified? If not can you take a look at it when you have the time.

**Pull request merged**